lib/Migrator/CMakeLists.txt

```diff
@@ -5,42 +5,23 @@ set(datafiles
   watchos4.json
   overlay4.json
 )
-set(SWIFTLIB_DIR
-    "${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/lib/swift")
-set(output_dir "${SWIFTLIB_DIR}/migrator")
 
-add_custom_command(
-    OUTPUT "${output_dir}"
-    COMMAND ${CMAKE_COMMAND} "-E" "make_directory" "${output_dir}")
-
-set(outputs)
-
-foreach(input ${datafiles})
-  set(source "${CMAKE_CURRENT_SOURCE_DIR}/${input}")
-  set(dest "${output_dir}/${input}")
-  if(CMAKE_SYSTEM_NAME STREQUAL Windows)
-    set(CMAKE_SYMLINK_COMMAND copy)
-  else()
-    set(CMAKE_SYMLINK_COMMAND create_symlink)
-  endif()
-
-  add_custom_command(OUTPUT
-                       "${output_dir}/${input}"
-                     DEPENDS
-                       "${CMAKE_CURRENT_SOURCE_DIR}/${input}"
-                     COMMAND
-                       "${CMAKE_COMMAND}" "-E" "${CMAKE_SYMLINK_COMMAND}" "${source}" "${dest}")
-  list(APPEND outputs "${output_dir}/${input}")
-endforeach()
-list(APPEND outputs "${output_dir}")
-
-add_custom_target("symlink_migrator_data"
-    DEPENDS "${output_dir}" "${outputs}"
-    COMMENT "Symlinking migrator data to ${output_dir}")
+add_custom_target(swift-migrator-data ALL
+                  COMMAND
+                    ${CMAKE_COMMAND} -E make_directory ${SWIFTLIB_DIR}/migrator
+                  COMMAND
+                    ${CMAKE_COMMAND} -E copy_if_different ${datafiles} ${SWIFTLIB_DIR}/migrator/
```

Comment:
This is the place where copy/create_symlink like I mentioned in one of the other comments was changed. We should be consistent here and then go through/audit them all and fix in 1 (subsequent) PR.

Reply:
Ugh, I just realized why I made the difference here.  It avoids having to loop over everything and having CMake just copy everything in a single shot.  Its simpler to read and should be faster due to fewer copies and call out to cmake.